Src/FluentAssertions/Equivalency/Steps/StructuralEqualityEquivalencyStep.cs

```diff
@@ -85,6 +82,13 @@ public class StructuralEqualityEquivalencyStep : IEquivalencyStep
                 where match is not null
                 select match;
 
+            // This excludes non-browsable members from the subject. If a member is marked non-browsable
+            // in the expectation, it gets excluded in GetMembersFromExpectation.
+            if (config.ExcludeNonBrowsable)
```

Comment:
> Implementing that as a IMemberSelectionRule won't work either, since any of them can readd that property.

Isn't it fine for a subsequent rule to readd a non-browsable member?

```cs
private class MyClass
{
    [EditorBrowsable(EditorBrowsableState.Never)]
    public int NonBrowsable { get; set; }

    public int Dummy { get; set; }
}

var subject = new { Dummy = 1, NonBrowsable = 1 };
var expected = new MyClass { Dummy = 1, NonBrowsable = 2 };

subject.Should().NotBeEquivalentTo(expected, opt => opt
    .ExcludingNonBrowsableMembers()
    .Including(e => e.NonBrowsable));
```

Reply:
If "include" and "exclude" are meant to be cumulative edits to the set and specifically refer to expectation members, then perhaps `ExcludingNonBrowsableMembers` isn't the right name for this functionality -- it shouldn't be lumped in with those.

What about renaming it to something like `SkipAllNonBrowsableMembers` so that there's no confusion about it being associated with "include" and "exclude" functions??

Reply:
Why do you think `ExcludingNonBrowsableMembers` isn't the right name?

Reply:
Because it currently operates as a blanket flag. It doesn't update the set cumulatively. If it's called, then at the end, any member that is marked non-browsable is excluded, even if you call `.Including` on it explicitly as shows in @jnyrup's example.